web/new_alt_text_manager.js

```diff
@@ -0,0 +1,396 @@
+/* Copyright 2023 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class NewAltTextManager {
+  #boundCancel = this.#cancel.bind(this);
+
+  #createAutomaticallyButton;
+
+  #currentEditor = null;
+
+  #cancelButton;
+
+  #descriptionContainer;
+
+  #dialog;
+
+  #disclaimer;
+
+  #firstTime = false;
+
+  #guessedAltText;
+
+  #isEditing = null;
+
+  #imagePreview;
+
+  #imageData;
+
+  #isAILoading = false;
+
+  #wasAILoading = false;
+
+  #learnMore;
+
+  #notNowButton;
+
+  #overlayManager;
+
+  #textarea;
+
+  #title;
+
+  #uiManager;
+
+  #previousAltText = null;
+
+  #telemetryData = null;
+
+  constructor(
+    {
+      descriptionContainer,
+      dialog,
+      imagePreview,
+      cancelButton,
+      disclaimer,
+      notNowButton,
+      saveButton,
+      textarea,
+      learnMore,
+      errorCloseButton,
+      createAutomaticallyButton,
+      title,
+    },
+    overlayManager
+  ) {
+    this.#cancelButton = cancelButton;
+    this.#createAutomaticallyButton = createAutomaticallyButton;
+    this.#descriptionContainer = descriptionContainer;
+    this.#dialog = dialog;
+    this.#disclaimer = disclaimer;
+    this.#notNowButton = notNowButton;
+    this.#imagePreview = imagePreview;
+    this.#textarea = textarea;
+    this.#learnMore = learnMore;
+    this.#title = title;
+    this.#overlayManager = overlayManager;
+
+    dialog.addEventListener("close", this.#close.bind(this));
```

Comment:
Should we use an `AbortController` to make sure that all event listeners get destroyed properly upon dialog close?

Reply:
This ctor is called only one time and set the different listeners for the different elements.
So removing them when closing the dialog implies to add them again when we reopen it: it isn't what we are doing in the current alt_text_manager.js and I followed the same pattern.
That said, I'm not sure to know what's the best approach. But for sure we can do that in a follow-up.